mne/stats/parametric.py

```diff
@@ -101,20 +95,76 @@ def f_oneway(*args):
     return _f_oneway(*args)[0]
 
 
-def _check_effects(effects):
-    """ Aux Function """
-    if effects.upper() not in defaults_twoway_rm['parse']:
-        raise ValueError('The value passed for `effects` is not supported. '
-                         'Please consider the documentation.')
-
-    return defaults_twoway_rm['parse'][effects]
+def _map_effects(n_factors, effects):
+    """Map effects to indices"""
+    if n_factors > len(ascii_uppercase):
+        raise ValueError('Maximum number of factors supported is 26')
+
+    factor_names = list(ascii_uppercase[:n_factors])
+
+    if isinstance(effects, string_types):
+        if '*' in effects and ':' in effects:
+            raise ValueError('Not "*" and ":" permitted in effects')
+        elif '+' in effects and ':' in effects:
+            raise ValueError('Not "+" and ":" permitted in effects')
+        elif effects == 'all':
+            effects = None
+        elif len(effects) == 1 or ':' in effects:
+            effects = [effects]
+        elif '+' in effects:
+            # all main effects
+            effects = effects.split('+')
+        elif '*' in effects:
+            pass  # handle later
+        else:
+            raise ValueError('"{0}" is not a valid option for "effects"'
+                             .format(effects))
+    if isinstance(effects, list):
+        bad_names = [e for e in effects if e not in factor_names]
+        if len(bad_names) > 1:
+            raise ValueError('Effect names: {0} are not valid. They should '
+                             'the first `n_factors` ({1}) characters from the'
+                             'alphabet'.format(bad_names, n_factors))
```

Comment:
@haribharadwaj I agree that this effects API could take some more documentation, as it can do some neat things for you that may not be that obvious. Let's see. No merge  blocker to me, we can improve incrementally.